src/cpp/session/modules/SessionReticulate.cpp

```diff
@@ -48,24 +48,19 @@ void updateReticulatePython(bool forInit)
    if (!forInit && s_pythonInitialized)
       return;
 
+   ASSERT_MAIN_THREAD;
```

Comment:
Here and elsewhere, some context we used to show in the warning message is now absent -- should we find a way to reintroduce that, or does the new message have enough context to fully remove the old one?

Reply:
IMHO we have enough context here.

I do think that if a thread is trying to figure out the default version of Python we should define and use an alternate thread-safe implementation, but given that I can't recall the exact context in which this was being touched by a separate thread we might want to keep the main-thread check to be safe.